docs: updated pattern template (#570)
carbon-design-system/carbon-website: docs/pattern-template.mdx

### Behaviors, structure, and functionality

- Describe pattern behaviors, including descriptions as well as any guidance on interactions and states.
- List out specific use cases, as well as situations where you would not use this pattern.
- Show the pattern in context. Include any user flows, and/or prototypes as needed.

### Best practices

List out best practices and include any design considerations that help with
making choices.

- Do's
- Dont's

### Pattern visuals

Different patterns require different visuals to best explain the behaviors and
hierarchy. Choose the options that make the most sense for the story you are
telling. See our [production guidelines](https://github.com/carbon-design-system/carbon-website/wiki/Production-guidelines) for information about creating assets for
the IBM Design System.

- Use annotated wireframes or sketches to explain the visual hierarchy of components.
- Create high-level user flows to explain the big picture.
- Create a functional prototype if that is the best way to illustrate the pattern's behavior.
- Provide a Sketch file with any symbols you've created in the development of this pattern.

> ### Variations

Reviewer: ```suggestion
### Variations
```
```suggestion
### Other use cases
```

> If there are variations that support different use cases, include those here

Reviewer: ```suggestion
If there are use cases that require a different solution, include those here
```

with corresponding explanations and visuals. Be sure to explain the reasons for

> using one pattern over another.

Reviewer: ```suggestion
using one solution over another.
```

## Related patterns and components

> Provide links to related patterns and components.

Reviewer: ```suggestion
Provide links to related patterns and components.
```
```suggestion
Which components did you use when building this pattern? List them here. 
```

If necessary, clarify any differences between this pattern and related patterns.

## Accessibility

Evaluate your pattern to ensure it meets [accessibility standards](/guidelines/accessibility/overview) and guidelines,
and provide details of compliance.

For example, “Users should be able to TAB into the input field of the search box
to being typing and press ENTER to run the search query.”

## References

> Include any citations for research, books, and articles you found helpful. These

Reviewer: ```suggestion
Help designers understand your process by explaining your rationale for the way you implemented the pattern. Include any research, citations, books or articles that you found helpful.  
```

> help designers understand your decisions.

Reviewer: ```suggestion
Include any material that will help designers understand your decisions.
```
